vulture_os/toolkit/api_parser/netskope/netskope.py

```diff
@@ -165,22 +167,40 @@ def execute(self):
                         log_ids.add(new_log["_event_id"])
                         logs.append(new_log)
                 offset += len(new_logs)
+                # Remove duplicated logs
+                i = 0
+                while i < len(new_logs):
+                    log = new_logs[i]
+                    if "_event_id" in log:
+                        if log["_event_id"] in log_ids:
+                            new_logs.pop(i)
+                            continue
+
+                        log_ids.add(log["_event_id"])
+
+                    if logtype == "alert":
+                        new_logs[i] = self.parse_alert_log(log)
+                    else:
+                        new_logs[i] = json.dumps(log)
+                    i += 1
+
+                logs.extend(new_logs)
```

Comment:
why are you trying to remove duplicated logs a second time?
at the end, `logs` is extended with both unparsed and parsed logs!